packages/@uppy/core/src/Uppy.js

```diff
@@ -583,6 +411,31 @@ class Uppy {
     }
   }
 
+  validateRestrictions (file, files = this.getFiles()) {
+    try {
+      this.#restricter.validate(file, files)
+      return { result: true }
+    } catch (err) {
+      return { result: false, reason: err.message }
+    }
+  }
+
+  checkRequiredMetaFieldsOnFile (file) {
+    const errorMap = this.#restricter.validateFile(file)
+
+    errorMap.forEach((_, error) => {
+      this.showOrLogErrorAndThrow(error, { file, showInformer: false, throwErr: false })
+    })
+
+    if (errorMap.size > 0) {
+      this.setFileState(file.id, { missingRequiredMetaFields: Array.from(errorMap.keys()) })
+    }
+  }
+
+  checkRequiredMetaFields (files) {
+    Object.values(files).forEach((file) => this.checkRequiredMetaFieldsOnFile(file))
+  }
```

Comment:
Note that I simplified this, it no longer has this: 

```js

    if (errors.length) {
      throw new AggregateRestrictionError(errors, `${this.i18n('missingRequiredMetaField')}`)
    }
```

I don't understand the purpose of `AggregateRestrictionError`, we have no tests for it, and it's never used in the codebase.

Reply:
The `AggregateError` allows to display one error instead of a dozens of errors. I think we should keep it, or at least provide a way not to overwhelm users if there are several errors with their files.

Reply:
I'm not sure. Currently things are being logged in `checkRequiredMetaFieldsOnFile` (which is called by `checkRequiredMetaFields`):

```js
    errorMap.forEach((_, error) => {
      this.showOrLogErrorAndThrow(error, { file, showInformer: false, throwErr: false })
    })
```

So all errors are already reported. If you throw an error it will be picked up and then all errors will be reported _again_. I think if the underlying function doesn't throw, then the top one shouldn't either, if it's merely a wrapper to iterate over files.

Reply:
As you can see `showInformer` is `false`, and `throwErr` is `false`, so I don't think that error would be reported with that call. IIRC it would save those errors in an aggregator and only report the aggregated error (unless there was one one error).

Reply:
The problem was, when looking deeper, we never aggregated those errors. Nor did it have the desired effect (obscured by `showOrLogErrorAndThrow`). I now emit all the files that failed as a normal restriction error. I will write some docs to clarify this. (TypeScript overloading would be great for this)